src/scripts/main.js

```diff
@@ -1 +1,65 @@
 'use strict';
+
+const successEl = (message) =>
+  `<div data-qa="notification" class="success">${message}</div>`;
+const errorEl = (message) =>
+  `<div data-qa="notification" class="error">${message}</div>`;
+
+const displayMessage = (message) => {
+  const body = document.querySelector('body');
+
+  body.insertAdjacentHTML('beforeend', message);
+};
+
+const promise1 = new Promise((resolve, reject) => {
+  const timeoutId = setTimeout(() => {
+    reject(errorEl('First promise was rejected'));
+  }, 3000);
+
+  const handleClick = () => {
+    clearTimeout(timeoutId);
+    resolve(successEl('First promise was resolved'));
+    document.removeEventListener('click', handleClick);
+  };
+
+  document.addEventListener('click', handleClick);
+});
+
+const promise2 = new Promise((resolve) => {
+  const handleClick = (e) => {
+    if (e.button === 0 || e.button === 2) {
+      resolve(successEl('Second promise was resolved'));
+      document.removeEventListener('click', handleClick);
+      document.removeEventListener('contextmenu', handleClick);
+    }
+  };
+
+  document.addEventListener('click', handleClick);
+  document.addEventListener('contextmenu', handleClick);
+});
+
+let leftClick = false;
+let rightClick = false;
```

Comment:
You have used two separate boolean variables `leftClick` and `rightClick` to track the click events for the third promise. This is fine, but you can improve the code by using a single variable. You can use a number variable `clickCount` and increment it on each click. When `clickCount` equals 2, you can resolve the promise.